### PR TITLE
feat(mojo): add mojo language

### PR DESCRIPTION
**Pull request by austincummings**
Copy over Python queries and extend with Mojo specific syntax.

### CODE REVIEW

**Comment by ribru17** [comment]
Instead of copying applicable queries, please use `; inherits: python` at the top of the file (unless maybe they aren't perfectly compatible)

**Comment by clason** [comment]
In particular, does the `mojo` _grammar_ inherit from Python? If not, why not?

**Comment by lucario387** [comment]
mojo-python is like typescript-javascript. All valid python codes are valid mojo code, not the other way around.

Seeing how the parser is written, I have some feeling there will be some decent breakage from python :? So keeping it separated will mean less maintanence effort

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1278,6 +1278,15 @@ list.mlir = {
   maintainers = { "@artagnon" },
 }
 
+list.mojo = {
+  install_info = {
+    url = "https://github.com/austincummings/tree-sitter-mojo",
+    files = { "src/parser.c", "src/scanner.c" },
+    branch = "main",
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1285]:
```suggestion
```
Only needed if it's not the default branch.

**Comment by clason** [comment]
> that's a valid concern or not though since I haven't worked much with this.

That _is_ a valid concern, and something you'd have to keep an eye on as parser (and query) maintainer.

**Comment by clason** [comment]
@austincummings please also test that `TSInstallFromGrammar mojo` works without errors. (`tree-sitter generate src/grammar.json`, basically)

File: queries/mojo/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+
+((call
+  function: (attribute
+    attribute: (identifier) @constructor))
+  (#lua-match? @constructor "^%u"))
+
+; Decorators
+((decorator
+  "@" @attribute)
+  (#set! "priority" 101))
```
Comment (lucario387) [inline queries/mojo/highlights.scm:62]: Wait, why?
If anything, move this down so you don't need priorities for it

File: queries/mojo/highlights.scm
Comment (lucario387) [inline queries/mojo/highlights.scm:1]: Can you note in this file (and others, if possible) which part are mojo-specific? That'd be helpful

File: queries/mojo/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+; from mojo
+(struct_definition
+  body: (block
+    (function_definition
+      name: (identifier) @function.method)))
+
+; This one causes a crash for some reason
+; (struct_definition
+;   superclasses: (argument_list
+;     (identifier) @type))
```
Comment (clason) [inline queries/mojo/highlights.scm:427]: 👀 

You might want to run the workflows from https://github.com/tree-sitter/workflows against your scanner, in particular.

**Comment by lucario387** [comment]
Whichever you think is more suitable for you and us.

**Comment by lucario387** [comment]
Is it ready for review now?

**Comment by clason** [comment]
@austincummings friendly ping

**Pull request closed without merging**